File: sol/1190.py
class Solution:
    def reverseParentheses(self, s: str) -> str:
        st = [[]]
        for c in s:
            if c == '(':
                st.append([])
            elif c != ')':
                st[-1].append(c)
            else:
                buff = st.pop()
                st[-1].extend(buff[::-1])
        return ''.join(st[-1])

File: sol/1190/1190.cpp
struct Solution {
    string reverseParentheses(string &s) {
        vector<string> st;
        st.push_back("");
        string buff{};
        for (char c : s){
            switch (c){
                case '(':
                    st.push_back("");
                    break;
                case ')':
                    buff = st.back();
                    st.pop_back();
                    reverse(buff.begin(), buff.end());
                    st.back() += buff;
                    break;
                default:
                    st.back().push_back(c);
                    break;
            }
        }
        return st.back();
    }
};
